<?xml version="1.0" encoding="UTF-8"?>
<testng-results ignored="0" total="8" passed="2" failed="1" skipped="5">
  <reporter-output>
  </reporter-output>
  <suite started-at="2025-05-04T15:47:12 IST" name="Default suite" finished-at="2025-05-04T15:48:44 IST" duration-ms="91655">
    <groups>
    </groups>
    <test started-at="2025-05-04T15:47:12 IST" name="Default test" finished-at="2025-05-04T15:48:44 IST" duration-ms="91655">
      <class name="FlipkartAutomation.Flipkart">
        <test-method is-config="true" signature="beforeTest()[pri:0, instance:FlipkartAutomation.Flipkart@8c3b9d]" started-at="2025-05-04T15:47:12 IST" name="beforeTest" finished-at="2025-05-04T15:47:47 IST" duration-ms="34940" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- beforeTest -->
        <test-method signature="openFlipkart()[pri:0, instance:FlipkartAutomation.Flipkart@8c3b9d]" started-at="2025-05-04T15:47:48 IST" name="openFlipkart" finished-at="2025-05-04T15:47:57 IST" duration-ms="8889" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- openFlipkart -->
        <test-method signature="checkIfLoggedIn()[pri:1, instance:FlipkartAutomation.Flipkart@8c3b9d]" started-at="2025-05-04T15:47:57 IST" name="checkIfLoggedIn" finished-at="2025-05-04T15:48:02 IST" depends-on-methods="FlipkartAutomation.Flipkart.openFlipkart" duration-ms="5446" status="SKIP">
          <exception class="org.testng.SkipException">
            <message>
              <![CDATA[Already logged in. Skipping login tests.]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.testng.SkipException: Already logged in. Skipping login tests.
at FlipkartAutomation.Flipkart.checkIfLoggedIn(Flipkart.java:87)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
            </full-stacktrace>
          </exception> <!-- org.testng.SkipException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- checkIfLoggedIn -->
        <test-method signature="clickLoginButton()[pri:2, instance:FlipkartAutomation.Flipkart@8c3b9d]" started-at="2025-05-04T15:48:02 IST" name="clickLoginButton" finished-at="2025-05-04T15:48:02 IST" depends-on-methods="FlipkartAutomation.Flipkart.checkIfLoggedIn" duration-ms="0" status="SKIP">
          <exception class="java.lang.Throwable">
            <message>
              <![CDATA[Method FlipkartAutomation.Flipkart.clickLoginButton() on instance FlipkartAutomation.Flipkart@8c3b9d depends on not successfully finished methods 
[FlipkartAutomation.Flipkart.checkIfLoggedIn() on instance FlipkartAutomation.Flipkart@8c3b9d]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.Throwable: Method FlipkartAutomation.Flipkart.clickLoginButton() on instance FlipkartAutomation.Flipkart@8c3b9d depends on not successfully finished methods 
[FlipkartAutomation.Flipkart.checkIfLoggedIn() on instance FlipkartAutomation.Flipkart@8c3b9d]
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:159)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.Throwable -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- clickLoginButton -->
        <test-method signature="mobileNumber()[pri:3, instance:FlipkartAutomation.Flipkart@8c3b9d]" started-at="2025-05-04T15:48:02 IST" name="mobileNumber" finished-at="2025-05-04T15:48:02 IST" depends-on-methods="FlipkartAutomation.Flipkart.clickLoginButton" duration-ms="8" status="SKIP">
          <exception class="java.lang.Throwable">
            <message>
              <![CDATA[Method FlipkartAutomation.Flipkart.mobileNumber() on instance FlipkartAutomation.Flipkart@8c3b9d depends on not successfully finished methods 
[FlipkartAutomation.Flipkart.clickLoginButton() on instance FlipkartAutomation.Flipkart@8c3b9d]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.Throwable: Method FlipkartAutomation.Flipkart.mobileNumber() on instance FlipkartAutomation.Flipkart@8c3b9d depends on not successfully finished methods 
[FlipkartAutomation.Flipkart.clickLoginButton() on instance FlipkartAutomation.Flipkart@8c3b9d]
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:159)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.Throwable -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- mobileNumber -->
        <test-method signature="requestOTPbtn()[pri:4, instance:FlipkartAutomation.Flipkart@8c3b9d]" started-at="2025-05-04T15:48:02 IST" name="requestOTPbtn" finished-at="2025-05-04T15:48:02 IST" depends-on-methods="FlipkartAutomation.Flipkart.mobileNumber" duration-ms="1" status="SKIP">
          <exception class="java.lang.Throwable">
            <message>
              <![CDATA[Method FlipkartAutomation.Flipkart.requestOTPbtn() on instance FlipkartAutomation.Flipkart@8c3b9d depends on not successfully finished methods 
[FlipkartAutomation.Flipkart.mobileNumber() on instance FlipkartAutomation.Flipkart@8c3b9d]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.Throwable: Method FlipkartAutomation.Flipkart.requestOTPbtn() on instance FlipkartAutomation.Flipkart@8c3b9d depends on not successfully finished methods 
[FlipkartAutomation.Flipkart.mobileNumber() on instance FlipkartAutomation.Flipkart@8c3b9d]
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:159)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.Throwable -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- requestOTPbtn -->
        <test-method signature="waitForManualOTPAndClickVerify()[pri:5, instance:FlipkartAutomation.Flipkart@8c3b9d]" started-at="2025-05-04T15:48:02 IST" name="waitForManualOTPAndClickVerify" finished-at="2025-05-04T15:48:02 IST" depends-on-methods="FlipkartAutomation.Flipkart.requestOTPbtn" duration-ms="0" status="SKIP">
          <exception class="java.lang.Throwable">
            <message>
              <![CDATA[Method FlipkartAutomation.Flipkart.waitForManualOTPAndClickVerify() on instance FlipkartAutomation.Flipkart@8c3b9d depends on not successfully finished methods 
[FlipkartAutomation.Flipkart.requestOTPbtn() on instance FlipkartAutomation.Flipkart@8c3b9d]]]>
            </message>
            <full-stacktrace>
              <![CDATA[java.lang.Throwable: Method FlipkartAutomation.Flipkart.waitForManualOTPAndClickVerify() on instance FlipkartAutomation.Flipkart@8c3b9d depends on not successfully finished methods 
[FlipkartAutomation.Flipkart.requestOTPbtn() on instance FlipkartAutomation.Flipkart@8c3b9d]
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:159)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
]]>
            </full-stacktrace>
          </exception> <!-- java.lang.Throwable -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- waitForManualOTPAndClickVerify -->
        <test-method signature="SearchProduct()[pri:6, instance:FlipkartAutomation.Flipkart@8c3b9d]" started-at="2025-05-04T15:48:02 IST" name="SearchProduct" finished-at="2025-05-04T15:48:33 IST" duration-ms="31308" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- SearchProduct -->
        <test-method signature="fetchProductDetails()[pri:7, instance:FlipkartAutomation.Flipkart@8c3b9d]" started-at="2025-05-04T15:48:33 IST" name="fetchProductDetails" finished-at="2025-05-04T15:48:44 IST" depends-on-methods="FlipkartAutomation.Flipkart.SearchProduct" duration-ms="10525" status="FAIL">
          <exception class="org.openqa.selenium.TimeoutException">
            <message>
              <![CDATA[Expected condition failed: waiting for presence of element located by: By.xpath: //div[contains(@class, '_1AtVbE')] (tried for 10 second(s) with 500 milliseconds interval)]]>
            </message>
            <full-stacktrace>
              <![CDATA[org.openqa.selenium.TimeoutException: Expected condition failed: waiting for presence of element located by: By.xpath: //div[contains(@class, '_1AtVbE')] (tried for 10 second(s) with 500 milliseconds interval)
at org.openqa.selenium.support.ui.WebDriverWait.timeoutException(WebDriverWait.java:84)
at org.openqa.selenium.support.ui.FluentWait.until(FluentWait.java:228)
at FlipkartAutomation.Flipkart.fetchProductDetails(Flipkart.java:159)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:141)
at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:687)
at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:230)
at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:63)
at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:995)
at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:203)
at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:154)
at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:134)
at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
at org.testng.TestRunner.privateRun(TestRunner.java:741)
at org.testng.TestRunner.run(TestRunner.java:616)
at org.testng.SuiteRunner.runTest(SuiteRunner.java:421)
at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:413)
at org.testng.SuiteRunner.privateRun(SuiteRunner.java:373)
at org.testng.SuiteRunner.run(SuiteRunner.java:312)
at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
at org.testng.TestNG.runSuitesSequentially(TestNG.java:1274)
at org.testng.TestNG.runSuitesLocally(TestNG.java:1208)
at org.testng.TestNG.runSuites(TestNG.java:1112)
at org.testng.TestNG.run(TestNG.java:1079)
at org.testng.remote.AbstractRemoteTestNG.run(AbstractRemoteTestNG.java:115)
at org.testng.remote.RemoteTestNG.initAndRun(RemoteTestNG.java:293)
at org.testng.remote.RemoteTestNG.main(RemoteTestNG.java:91)
Caused by: org.openqa.selenium.NoSuchElementException: Unable to locate element: //div[contains(@class, '_1AtVbE')]
For documentation on this error, please visit: https://www.selenium.dev/documentation/webdriver/troubleshooting/errors#no-such-element-exception
Build info: version: '4.30.0', revision: '509c7f17cc*'
System info: os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '23.0.1'
Driver info: org.openqa.selenium.firefox.FirefoxDriver
Command: [a710e32e-60e0-475e-bef1-78d15b3daef8, findElement {using=xpath, value=//div[contains(@class, '_1AtVbE')]}]
Capabilities {acceptInsecureCerts: true, browserName: firefox, browserVersion: 133.0.3, moz:accessibilityChecks: false, moz:buildID: 20241209150345, moz:geckodriverVersion: 0.36.0, moz:headless: false, moz:platformVersion: 10.0, moz:processID: 10644, moz:profile: C:\Users\PRATHA~1\AppData\L..., moz:shutdownTimeout: 60000, moz:webdriverClick: true, moz:windowless: false, pageLoadStrategy: normal, platformName: windows, proxy: Proxy(), setWindowRect: true, strictFileInteractability: false, timeouts: {implicit: 0, pageLoad: 300000, script: 30000}, unhandledPromptBehavior: dismiss and notify, userAgent: Mozilla/5.0 (Windows NT 10....}
Session ID: a710e32e-60e0-475e-bef1-78d15b3daef8
at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:62)
at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:501)
at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:485)
at org.openqa.selenium.remote.ErrorCodec.decode(ErrorCodec.java:167)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:138)
at org.openqa.selenium.remote.codec.w3c.W3CHttpResponseCodec.decode(W3CHttpResponseCodec.java:50)
at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:215)
at org.openqa.selenium.remote.service.DriverCommandExecutor.invokeExecute(DriverCommandExecutor.java:216)
at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:174)
at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:545)
at org.openqa.selenium.remote.ElementLocation$ElementFinder$2.findElement(ElementLocation.java:165)
at org.openqa.selenium.remote.ElementLocation.findElement(ElementLocation.java:59)
at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:368)
at org.openqa.selenium.remote.RemoteWebDriver.findElement(RemoteWebDriver.java:362)
at org.openqa.selenium.support.ui.ExpectedConditions$6.apply(ExpectedConditions.java:171)
at org.openqa.selenium.support.ui.ExpectedConditions$6.apply(ExpectedConditions.java:168)
at org.openqa.selenium.support.ui.FluentWait.until(FluentWait.java:203)
... 27 more
]]>
            </full-stacktrace>
          </exception> <!-- org.openqa.selenium.TimeoutException -->
          <reporter-output>
          </reporter-output>
        </test-method> <!-- fetchProductDetails -->
        <test-method is-config="true" signature="afterTest()[pri:0, instance:FlipkartAutomation.Flipkart@8c3b9d]" started-at="2025-05-04T15:48:44 IST" name="afterTest" finished-at="2025-05-04T15:48:44 IST" duration-ms="14" status="PASS">
          <reporter-output>
          </reporter-output>
        </test-method> <!-- afterTest -->
      </class> <!-- FlipkartAutomation.Flipkart -->
    </test> <!-- Default test -->
  </suite> <!-- Default suite -->
</testng-results>
